engine/consensus/matching/engine.go

```diff
@@ -588,34 +600,74 @@ func (e *Engine) checkingSealing() error {
 
 	// request execution receipts for unsealed finalized blocks
 	requestReceiptsSpan := e.tracer.StartSpanFromParent(sealingSpan, trace.CONMatchCheckSealingRequestPendingReceipts)
-	err = e.requestPendingReceipts()
+	pendingReceiptsRequested, err := e.requestPendingReceipts()
 	requestReceiptsSpan.Finish()
 	if err != nil {
 		return fmt.Errorf("could not request pending block results: %w", err)
 	}
 
 	// request result approvals for pending incorporated results
 	requestApprovalsSpan := e.tracer.StartSpanFromParent(sealingSpan, trace.CONMatchCheckSealingRequestPendingApprovals)
-	err = e.requestPendingApprovals()
+	pendingApprovalsRequested, err := e.requestPendingApprovals()
 	requestApprovalsSpan.Finish()
 	if err != nil {
 		return fmt.Errorf("could not request pending result approvals: %w", err)
 	}
 
+	mempoolHasNextSeal := false
+
+	// check if we end up created a seal in mempool for the next unsealed block
+	for _, unsealed := range nextUnsealeds {
+		_, mempoolHasNextSeal = e.seals.ByID(unsealed.IncorporatedResultID)
+		if mempoolHasNextSeal {
+			break
+		}
+	}
+
+	lg.Info().
+		Int("sealed_block_count", len(sealedBlockIDs)).
+		Str("duration", time.Since(startTime).String()).
+		Bool("mempool_has_next_seal", mempoolHasNextSeal).
+		Int("pending_receipts_requested", pendingReceiptsRequested).
+		Int("pending_approvals_requested", pendingApprovalsRequested).
+		Msg("checking sealing finished successfully")
+
 	return nil
 }
 
+func nextUnsealedID(state protocol.State) (flow.Identifier, bool) {
+	lastSealed, err := state.Sealed().Head()
+	if err != nil {
+		return flow.ZeroID, false
+	}
```

Comment:
This would be an indication of a serious bug. We always have a sealed result. Even if it the result for the root block. But just ignoring a symptom of a fatal bug and pretending returning some dummy value is not a good style in my opinion. 

#### Suggestion
* lets add a comment why this should never happen:
  ```
   // this should never happen, as we always have a sealed result (at first boot, the result for the root block)
  ```
* panic if it happens nevertheless